crates/teloxide-macros/src/fields_parse.rs

```diff
@@ -139,12 +139,12 @@ fn parser_with_separator<'a>(
                 let res = #res;
 
                 match splitted.next() {
-                    Some(d) => ::std::result::Result::Err(teloxide::utils::command::ParseError::TooManyArguments {
+                    Some(d) if !s.is_empty() => ::std::result::Result::Err(teloxide::utils::command::ParseError::TooManyArguments {
```

Comment:
Actually, this will behave weirdly for something like `sep = '|'; s = "|a"` (expected 0 arguments, this will return `Ok`, actually there are 2 arguments)

Reply:
It checks the original string, not the next argument